st2actions/st2actions/policies/concurrency.py

```diff
@@ -61,9 +61,10 @@ def _apply_before(self, target):
                       count, target.action, self._policy_ref, action)
             status = self._get_status_for_policy_action(action=self.policy_action)
 
-        # Update the status in the database and publish status so the appropriate runner can
-        # cancel the execution appropriately.
-        target = action_service.update_status(target, status, publish=True)
+        # Update the status in the database. Publish status for cancellation so the
+        # appropriate runner can cancel the execution.
+        publish = (status == action_constants.LIVEACTION_STATUS_CANCELING)
```

Comment:
Heh, so `publish` was intentionally set to `False` before previous change. That makes sense now :)